pandas/_libs/tslibs/period.pyx

```diff
@@ -1058,18 +1058,20 @@ cdef class _Period(object):
         return hash((self.ordinal, self.freqstr))
 
     def _add_delta(self, other):
-        if isinstance(other, (timedelta, np.timedelta64, offsets.Tick)):
-            offset = frequencies.to_offset(self.freq.rule_code)
-            if isinstance(offset, offsets.Tick):
```

Comment:
This along with the line above is redundant; `isinstance(offset, offsets.Tick)` if and only if `isinstance(self.freq, offsets.Tick)`.